### DISCUSSION TITLE
Best way to enable ansi formatting for terminal output only

### DISCUSSION THREAD

**Discussion by aaronriekenberg** [tokio-rs/tracing#2640] [Q&A] [answered]
I have an application using 

```rust
tracing_subscriber::fmt::init();
```

Have noticed by default this enables ansi formatting, which has formatting issues when redirecting output to a file or running as a systemd service.

I came up with the below function to do basically the equivalent of above but only enable ansi formatting if `std::io::stdout().is_terminal()` is true.  This is a new feature in rust 1.70.0: https://blog.rust-lang.org/2023/06/01/Rust-1.70.0.html#isterminal

Is there a simpler way to do what is below?  Could something similar to below be considered for the default `tracing_subscriber::fmt::init()` to enable ansi formatting only if outputting to a terminal?

Thanks!

```rust
fn initialize_tracing_subscriber() {
    use tracing_subscriber::{filter::LevelFilter, fmt, prelude::*, EnvFilter};

    use std::io::IsTerminal;

    let use_ansi = std::io::stdout().is_terminal();

    let env_filter = EnvFilter::builder()
        .with_default_directive(LevelFilter::INFO.into())
        .from_env_lossy();

    tracing_subscriber::registry()
        .with(env_filter)
        .with(fmt::layer().with_ansi(use_ansi))
        .init();
}
```

**Comment by davidbarsky** [comment, marked as answer]
Sorry for the delay. What you wrote is generally the correct approach, and while we'd like to make `tracing_subscriber::fmt` call `std::io::stdout().is_terminal()`, we can't due to our current MSRV.
